internal/namespace/nsutils/nsutils.go

```diff
@@ -62,16 +62,27 @@ func DisplayNameSlice(ns *objectcontext.ObjectContext) []string {
 	return nsArr
 }
 
-// LabelsAnnotationsBasedOnParent returns labels and annotations for a namespace,
-// the labels and annotations are based on the ones of the namespace of the parent.
-func LabelsAnnotationsBasedOnParent(parentNS *objectcontext.ObjectContext, nsName string) (map[string]string, map[string]string) {
-	childNamespaceDepth := strconv.Itoa(Depth(parentNS.Object) + 1)
-	parentDisplayName := DisplayName(parentNS.Object)
+// LabelsBasedOnParent returns labels for a namespace based on the ones of the namespace of the parent.
+func LabelsBasedOnParent(parentNS *objectcontext.ObjectContext, nsName string) map[string]string {
+	parentDisplayNameSliced := DisplayNameSlice(parentNS)
 
 	labels := make(map[string]string)
 	labels[danav1.Parent] = parentNS.Object.(*corev1.Namespace).Name
 	labels[danav1.Hns] = "true"
 
+	for _, ns := range parentDisplayNameSliced {
+		labels[ns] = "true"
```

Comment:
This label is kind of confusing, may add it as
 `<parentNS>/parent : "true"` or something like that?
Not sure just thinking out loud

Reply:
This is not possible due to the label name constraints. Label keys have to be up to [63 characters](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names); Namespace names are have the same constraint so we can't have anything other than the name itself in the label key.